Wikipedia/Code/WikidataDescriptionEditingController.swift

```diff
@@ -39,14 +39,44 @@ enum WikidataPublishingError: LocalizedError {
 }
 
 @objc public final class WikidataDescriptionEditingController: NSObject {
-    private var blacklistedLanguages = Set<String>()
+    weak var viewContext: NSManagedObjectContext?
+    private let semaphore = DispatchSemaphore(value: 1)
+
+    @objc public init(with viewContext: NSManagedObjectContext) {
+        self.viewContext = viewContext
+    }
+
+    private let BlacklistedLanguagesKey = "WMFWikidataDescriptionEditingBlacklistedLanguagesKey"
+    private var blacklistedLanguages: NSSet {
+        assert(Thread.isMainThread)
+        semaphore.wait()
+        defer {
+            semaphore.signal()
+        }
+        let fallback = NSSet(set: ["en"])
+        guard
+            let viewContext = viewContext,
+            let keyValue = viewContext.wmf_keyValue(forKey: BlacklistedLanguagesKey),
+            let value = keyValue.value as? NSSet else {
+                assertionFailure("Failed to retrieve WMFKeyValue for key \(BlacklistedLanguagesKey), returning fallback object \(fallback)")
+                return fallback
+        }
+        return value
+    }
 
     @objc public func setBlacklistedLanguages(_ blacklistedLanguagesFromRemoteConfig: Array<String>) {
-        blacklistedLanguages = Set(blacklistedLanguagesFromRemoteConfig)
+        assert(Thread.isMainThread)
+        assert(viewContext != nil)
+        semaphore.wait()
```

Comment:
If this & the above are always on the main thread, the semaphore shouldn't be needed